### PR TITLE
Add functionality for testing and releasing packages

### PR DESCRIPTION
**Pull request by mdonnalley**
@W-7824758@

### CODE REVIEW

File: src/commands/sign-npm-package.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  - run:
+      name: Set credentials
+      command: |
+        echo //registry.npmjs.org/:_authToken=<<parameters.npm_token>> > /root/.npmrc
+        echo unsafe-perm = true >> /root/.npmrc
+        echo <<parameters.salesforce_key>> | base64 --decode > <<parameters.key_path>>
+  - run:
+      name: Sign npm package
+      command: |
+        export SFDX_DEVELOPER_TRUSTED_FINGERPRINT=22:F4:87:2C:B2:22:04:1E:F5:D2:AB:AE:E7:69:81:10:1E:C1:F7:0B
```
Comment (amphro) [inline src/commands/sign-npm-package.yml:33]: Weren't you also going to have this be a parameter too with a default? That way if it changes from under us we can still get a job to pass quickly without updating the orb?

File: src/commands/sign-npm-package.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        echo $RESULT | jq .
+
+        export FILE_NAME=$(echo $RESULT | jq -r '.result.filename')
+        export TARPATH=$(echo $RESULT | jq -r '.result.tarPath')
+        export PKG_NAME=$(echo $RESULT | jq -r '.result.name')
+        export PKG_VERSION=$(echo $RESULT | jq -r '.result.version')
+
+        aws s3 cp $FILE_NAME s3://dfc-data-production/media/salesforce-cli/signatures/
+
+        npm publish $TARPATH
```
Comment (amphro) [inline src/commands/sign-npm-package.yml:48]: We will also have to publish on different tags as well. I'm assuming this is the latest tag by default right? I would love to have a parameter to this where people could set the tag in their job, either by using the branch name if they want or hard coding a specific branch to a specific tag.

File: src/examples/test-and-release.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        - release-management/test-package:
+            name: node-latest
+            node_version: latest
+        - release-management/test-package:
+            name: node-12
+        - release-management/release-package:
+            context: AWS
+            salesforce_key: $SALESFORCE_KEY
+            npm_token: $NPM_TOKEN
+            key_path: /root/salesforce-cli.key
```
Comment (amphro) [inline src/examples/test-and-release.yml:26]: Since default, let's remove it from the example?

File: src/examples/test-and-release.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    version: 2
+    test-and-release:
+      jobs:
+        - release-management/test-package:
+            name: node-latest
+            node_version: latest
+        - release-management/test-package:
+            name: node-12
+        - release-management/release-package:
+            context: AWS
```
Comment (amphro) [inline src/examples/test-and-release.yml:23]: What is this? I don't see a context parameter in the orb.

File: src/examples/test-and-release.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            npm_token: $NPM_TOKEN
+            key_path: /root/salesforce-cli.key
+            requires:
+              - node-latest
+            filters:
+              tags:
+                only: /^v.*/
+              branches:
+                ignore: /.*/
+        - release-management/cache:
```
Comment (amphro) [inline src/examples/test-and-release.yml:34]: As mentioned in the config plugin, I think we should remove this and handle cache for them behind the scenes by just doing it after the install.

File: src/jobs/release-package.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+steps:
+  - checkout
+  - restore_cache:
+      keys:
+        - v1-npm-{{checksum ".circleci/config.yml"}}-{{checksum "yarn.lock"}}
+        - v1-npm-{{checksum ".circleci/config.yml"}}
+  - run:
+      name: Install dependencies
+      command: yarn
```
Comment (amphro) [inline src/jobs/release-package.yml:37]: For example with the cache, right after this yarn command you would save the cache.

File: src/jobs/test-package.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+steps:
+  - checkout
+  - restore_cache:
+      keys:
+        - v1-npm-{{checksum ".circleci/config.yml"}}-{{checksum "yarn.lock"}}
+        - v1-npm-{{checksum ".circleci/config.yml"}}
+  - run:
+      name: Install dependencies
+      command: yarn
```
Comment (amphro) [inline src/jobs/test-package.yml:27]: Same here, right after you would save the cache.

File: src/jobs/test-package.yml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+executor:
+  name: default
+  tag: << parameters.node_version >>
+
+steps:
+  - checkout
+  - restore_cache:
+      keys:
+        - v1-npm-{{checksum ".circleci/config.yml"}}-{{checksum "yarn.lock"}}
```
Comment (amphro) [inline src/jobs/test-package.yml:23]: You allow them to specify the package manager in the publish but not in these scripts. Is that something we want to support?

File: src/examples/test-and-release.yml (hunk 4 above)
Comment (mdonnalley) [reply src/examples/test-and-release.yml:23]: Contexts let you define a set of environment variables https://circleci.com/docs/2.0/contexts/

File: src/jobs/test-package.yml (hunk 8 above)
Comment (mdonnalley) [reply src/jobs/test-package.yml:23]: @amphro hm, that's a good point. Now that I'm thinking about it, I think we should force people into using `yarn`. We can support `npm` down the road if we need to

Comment (amphro) [reply src/jobs/test-package.yml:23]: Sounds good to me.

**Pull request merged**